src/main/kotlin/app/revanced/patches/youtube/layout/autocaptions/patch/AutoCaptionsPatch.kt

```diff
@@ -0,0 +1,76 @@
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.impl.BytecodeData
+import app.revanced.patcher.extensions.addInstructions
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.annotations.DependsOn
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patcher.patch.impl.BytecodePatch
+import app.revanced.patches.youtube.layout.autocaptions.annotations.autocaptionsCompatibility
+import app.revanced.patches.youtube.layout.autocaptions.fingerprints.StartVideoInformerFingerprint
+import app.revanced.patches.youtube.layout.autocaptions.fingerprints.SubtitleButtonControllerFingerprint
+import app.revanced.patches.youtube.layout.autocaptions.fingerprints.SubtitleTrackFingerprint
+import app.revanced.patches.youtube.misc.integrations.patch.IntegrationsPatch
+import app.revanced.patches.youtube.misc.settings.bytecode.patch.SettingsPatch
+import app.revanced.patches.youtube.misc.settings.framework.components.impl.StringResource
+import app.revanced.patches.youtube.misc.settings.framework.components.impl.SwitchPreference
+
+@Patch
+@DependsOn([IntegrationsPatch::class, SettingsPatch::class])
+@Name("hide-autoplay-button")
+@Description("Hides the autoplay button in the video player.")
+@autocaptionsCompatibility
+@Version("0.0.1")
+class AutoCaptionsPatch : BytecodePatch(
+    listOf(
+        StartVideoInformerFingerprint
+    )
+) {
+    override fun execute(data: BytecodeData): PatchResult {
+        SettingsPatch.PreferenceScreen.LAYOUT.addPreferences(
+            SwitchPreference(
+                "revanced_autocaptions_enabled",
+                StringResource("revanced_autocaptions_enabled_title", "Disable forced captions"),
+                false,
+                StringResource("revanced_autocaptions_summary_on", "Forced captions is enabled"),
+                StringResource("revanced_autocaptions_summary_off", "Forced captions is disabled")
+            )
+        )
+
+        val startVideoInformerMethod = StartVideoInformerFingerprint.result!!.mutableMethod
+
+        startVideoInformerMethod.addInstructions(
+            0, """
+            const/4 v0, 0x0
+            sput-boolean v0, Lapp/revanced/integrations/patches/ForceDisableCaptionsPatch;->disablecaption:Z
+        """
+        )
+
+        val subtitleButtonControllerMethod = SubtitleButtonControllerFingerprint.result!!.mutableMethod
+
+        subtitleButtonControllerMethod.addInstructions(
+            0, """
+            const/4 v0, 0x1
+            sput-boolean v0, Lapp/revanced/integrations/patches/ForceDisableCaptionsPatch;->disablecaption:Z
+        """
+        )
+
+        val subtitleTrackMethod = SubtitleTrackFingerprint.result!!.mutableMethod
+
+        subtitleTrackMethod.addInstructions(
+            0, """
+            sget-boolean v0, Lapp/revanced/integrations/patches/ForceDisableCaptionsPatch;->disablecaption:Z
```

Comment:
Instead of using this field, you should use the following method:
https://github.com/revanced/revanced-integrations/pull/133/files#diff-b4827a27e50e6dd92f95150c7f3e3d504144dc8d2c611672ad7be74eb962e149R9-R9

Reply:
This boolean is needed to track the video player, once is shown. This patch is not always enabled, but it's only for a little amount of time before the video player is shown. This is a trick to prevent subtitles from activating.

Reply:
So what is `captionsEnabled()` used for?

Reply:
Anyway...this boolean it's not related to toggle. I'll add this after I ensure that patch is working.

Reply:
> So what is `captionsEnabled()` used for?

Should be needed to track the toggle state (true or false). As I said, I still need to add it to patch, but I ensure this is working before.

Reply:
>  As I said, I still need to add it to patch

I dont see where, anyways, you should use that method so the captions setting can be toggled via the settings.

Reply:
> > As I said, I still need to add it to patch
> 
> I dont see where, anyways, you should use that method so the captions setting can be toggled via the settings.

I haven't used it yet. :)

Wait...